kong/pdk/tracing.lua

```diff
@@ -158,7 +158,7 @@ local function create_span(tracer, options)
       or generate_trace_id()
   local sampled = span.parent and span.parent.should_sample
       or options.should_sample
-      or tracer and tracer.sampler(trace_id)
+      or not span.parent and tracer and tracer.sampler(trace_id)
```

Comment:
this means that if the parent span had disabled sampling, we set sampled to false in this span too, using a parent-based sampler with a fallback to the probability sampler.
The previous logic meant we would only inherit the sampling flag from the parent if it was "true"

Reply:
maybe. Right now they are grouped "by line" which I think make the whole logic quite readable. I will try to see if parenthesis can be used to improve this.